deploy.py

```diff
@@ -432,14 +436,23 @@ def create_tab_file(self, binaries):
     ]
     if self.args.verbose_build:
       elf2tab_args.append("--verbose")
+    stack_sizes = []
     for arch, app_file in binaries.items():
       dest_file = os.path.join(self.tab_folder, "{}.elf".format(arch))
       shutil.copyfile(app_file, dest_file)
       elf2tab_args.append(dest_file)
+      # extract required stack size directly from binary
+      nm = self.checked_command_output(
+          ["nm", "--print-size", "--radix=x", app_file])
+      for line in nm.splitlines():
+        if "STACK_MEMORY" in line:
+          required_stack_size = int(line.split(" ", maxsplit=2)[1], 16)
+          stack_sizes.append(required_stack_size)
 
     elf2tab_args.extend([
-        "--stack={}".format(STACK_SIZE), "--app-heap={}".format(APP_HEAP_SIZE),
-        "--kernel-heap=1024", "--protected-region-size=64"
+        "--stack={}".format(max(stack_sizes)),
```

Comment:
We can have more than one compiled app/example in the TAB file when we'll start to compile for multiple architectures (armv7, armv8 and risc-v come to mind).
But the stack size should remain the same and using a `set()` instead of a `list()` and warning if we have more than one entry would make sense.

Reply:
It's a set now and prints an error if different stack sizes are detected.

Reply:
Oh I missed the outer for loop over archs. SGTM